src/land/madis_snow2ioda.py

```diff
@@ -0,0 +1,165 @@
+
+        sites = np.empty_like(vals, dtype=object)
+        for i in range(len(vals[:])):
+            sites[i] = sids[i]
+        times = np.empty_like(vals, dtype=np.int64)
+        times = obst
+
+        vals = vals.astype('float32')
+        lats = lats.astype('float32')
+        lons = lons.astype('float32')
+        elvs = elvs.astype('float32')
+        errs = 0.0*vals
+        errs = errs.astype('float32')
+        qflg = qflg.astype('int32')
+
+        # use maskout options
+        if self.mask:
+            with np.errstate(invalid='ignore'):
+                mask = (vals >= 0.0) & (vals < 2000.0)
+            vals = vals[mask]
+            errs = errs[mask]
+            qflg = qflg[mask]
+            lons = lons[mask]
+            lats = lats[mask]
+            elvs = elvs[mask]
+            sites = sites[mask]
+            times = times[mask]
+
+        for i in range(len(vals)):
+            if vals[i] >= 0.0:
+                errs[i] = 40.0
+            else:
+                errs[i] = _FillValue
+
+        # add metadata variables
+        self.outdata[('dateTime', 'MetaData')] = times
+        self.outdata[('stationIdentification', 'MetaData')] = sites
+        self.outdata[('latitude', 'MetaData')] = lats
+        self.outdata[('longitude', 'MetaData')] = lons
+        self.outdata[('stationElevation', 'MetaData')] = elvs
+        self.varAttrs[('stationElevation', 'MetaData')]['units'] = 'm'
+        # self.varAttrs[('dateTime', 'MetaData')]['units'] = iso8601_string
+        self.varAttrs[('dateTime', 'MetaData')]['_FillValue'] = long_missing_value
+
+        for iodavar in ['totalSnowDepth']:
+            self.outdata[self.varDict[iodavar]['valKey']] = vals
+            self.varAttrs[self.varDict[iodavar]['valKey']]['_FillValue'] = float_missing_value
+            self.outdata[self.varDict[iodavar]['errKey']] = errs
+            self.varAttrs[self.varDict[iodavar]['errKey']]['_FillValue'] = float_missing_value
+            self.outdata[self.varDict[iodavar]['qcKey']] = qflg
+
+        DimDict['Location'] = len(self.outdata[('dateTime', 'MetaData')])
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description=('Read MADIS snow depth file(s) and Converter'
+                     ' of native NetCDF format for observations of total'
+                     ' snow depth to IODA netCDF format.')
+    )
+    parser.add_argument('-i', '--input',
+                        help="name of madis total snow depth input file(s)",
+                        type=str, required=True)
+    parser.add_argument('-o', '--output',
+                        help="name of ioda output file",
+                        type=str, required=True)
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument('--maskMissing', dest='mask',
+                          help="switch to mask missing values: default=False",
+                          default=False, action='store_true', required=False)
+
```

Comment:
@YoulongXia-NOAA I just had to change this to a logical switch,,, if present do masking,,, if not present do nothing (the old `default`)

Reply:
@YoulongXia-NOAA will go ahead and merge if you are aware and OK with this change

Reply:
@BenjaminRuston, it is fine for me. Please merge it.